### PR TITLE
Rxjava 3 Migration

### PR DESCRIPTION
**Pull request by raghulvaikundam**
Migrated the rxjava from version 2 to version 3

### CODE REVIEW

**Comment by VaiTon** [comment]
Have you tested this? This should not work with okhttp 3.13. From changelog:

> Version 2.7.0 (2019-12-09)
> 
> This release changes the minimum requirements to Java 8+ or Android 5+. See this blog post for more information on the change.
> 
>     New: Upgrade to OkHttp 3.14.4. Please see its changelog for 3.x.
>     Fix: Allow service interfaces to extend other interfaces.
>     Fix: Ensure a non-null body is returned by Response.error.
>

**Comment by raghulvaikundam** [comment]
> This release changes the minimum requirements to Java 8+ or Android 5+. See this blog post for more information on the change.

Yes I tested it. It is working.

**Comment by VaiTon** [comment]
As i said, from the retrofit changelog: https://github.com/square/retrofit/blob/master/CHANGELOG.md#version-290-2020-05-20

> 
> Version 2.7.0 (2019-12-09)
> 
> This release changes the minimum requirements to Java 8+ or Android 5+. See this blog post for more information on the change.
> 
>     New: Upgrade to OkHttp 3.14.4. Please see its changelog for 3.x.
>     Fix: Allow service interfaces to extend other interfaces.
>     Fix: Ensure a non-null body is returned by Response.error.
>

**Comment by VaiTon** [comment]
Have you tested on emulator?

**Comment by raghulvaikundam** [comment]
> Have you tested on emulator?

yes tested in emulator. It is working. Now I have also updated the retrofit version to 2.9.0

**Comment by raghulvaikundam** [comment]
@VaiTon is the change good enough to merge?

**Comment by VaiTon** [comment]
I would like the comment of @teolemon on upgrading the minSdk

**Comment by teolemon** [comment]
This sounds like something for Release +1. I'd like to focus on releasing the current master once it's stabilised. Excluding users from the update would be a little sad, especially since we've bumped any non-breaking library we could.
Breaking changes will be for +1.

**Comment by teolemon** [comment]
https://github.com/ReactiveX/RxJava/wiki/What's-different-in-3.0

**Comment by naivekook** [comment]
Hi @raghulvaikundam!

Any updates here? I can pick it up and help with resolving merge conflicts

File: app/build.gradle.kts [hunk 1]
```diff
@@ -57,19 +57,19 @@ dependencies {
...
     //Rx
-    implementation("io.reactivex.rxjava2:rxjava:2.2.19")
-    implementation("io.reactivex.rxjava2:rxandroid:2.1.1")
+    implementation("io.reactivex.rxjava3:rxjava:3.0.0")
+    implementation("io.reactivex.rxjava3:rxandroid:3.0.0")
 
     //Networking
-    implementation("com.squareup.retrofit2:retrofit:2.6.4")
+    implementation("com.squareup.retrofit2:retrofit:2.9.0")
     implementation("com.squareup.retrofit2:converter-jackson:2.6.4")
```
Comment (naivekook) [inline app/build.gradle.kts:71]: Should we update it to 2.9.0?

File: app/build.gradle.kts [hunk 2]
```diff
@@ -57,19 +57,19 @@ dependencies {
...
-    implementation("com.squareup.retrofit2:adapter-rxjava2:2.6.4")
+    implementation("com.squareup.retrofit2:adapter-rxjava3:2.9.0")
     implementation("com.squareup.okhttp3:logging-interceptor:3.12.11")
```
Comment (naivekook) [inline app/build.gradle.kts:73]: I guess we have to update it too because retrofit uses newer version of okhttp

**Comment by raghulvaikundam** [comment]
> Hi @raghulvaikundam!
> 
> Any updates here? I can pick it up and help with resolving merge conflicts

I waited as it was deferred for the next release. If it is good to go, I can start resolving the conflicts

**Comment by VaiTon** [comment]
> 
> 
> > Hi @raghulvaikundam!
> > Any updates here? I can pick it up and help with resolving merge conflicts
> 
> I waited as it was deferred for the next release. If it is good to go, I can start resolving the conflicts

@teolemon what do you think? Remember the api bump we have to do

**Comment by teolemon** [comment]
Closing as it will be easier to redo completely, due to the Kotlin migration.
Really sorry about that @raghulvaikundam

**Pull request closed without merging**